### PR TITLE
KFSPTS-25321 upgrade to the 1/05/22 version of financials

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by db483** [comment]
retest this please

**Review by cah292 — CHANGES_REQUESTED** [review]
This is still under review, but so far I've found a few items that may need to be addressed.

File: pom.xml
Comment (db483) [reply pom.xml:520]: removing

File: pom.xml
Comment (db483) [reply pom.xml:267]: rebased and updating

Comment (db483) [reply pom.xml:267]: rebased and updated

File: pom.xml
Comment (db483) [reply pom.xml:520]: updated

File: src/main/java/org/kuali/kfs/kim/impl/role/RoleServiceImpl.java
Comment (db483) [reply src/main/java/org/kuali/kfs/kim/impl/role/RoleServiceImpl.java:179]: updated

File: src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml:151]: What is the current plan for updating the various doc types to include this new ObjectCode route node? Would you prefer using the workflow XML provided under the base financials `/kfs-core/src/main/resources/workflow-upgrades/2021-12/` folder instead of SQL, and if so should that file or its contents be included in the cu-kfs project?

**Review by cah292 — CHANGES_REQUESTED** [review]
I found some more sections needing cleanup, but overall the changes look good.

Also, given our inconsistencies with the usage of KualiCo license headers in our files, we may want to create a follow-up user story to prepare a more consistent license header setup across the cu-kfs project.

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:1070]: updated

File: src/main/resources/edu/cornell/kfs/vnd/cu-spring-vnd.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/vnd/cu-spring-vnd.xml:29]: good point, I will revert to merge true

File: src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java:96]: updated

File: src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java:137]: updated

File: src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java:323]: Good question. I created https://jira.cornell.edu/browse/KFSPTS-26565.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:33]: Good question. The old base code was only setting Content-type but the new base code is setting both the Accept header and the Content-type which is why I updated our customization to set both. Since the unit tests pass I thought the punchout service worked fine but I may have missed something.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:40]: updated though documentation seems to suggest that ISO-8859-1 is the default charset: https://hc.apache.org/httpcomponents-core-5.1.x/current/httpcore5/apidocs/org/apache/hc/core5/http/io/entity/StringEntity.html#StringEntity(java.lang.String,%20java.nio.charset.Charset)

File: src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml:151]: Excellent question. Functionals would like to first better understand the ObjectCode routing and then determine te best time to implement locally. There is a jira in the next Sprint for this: https://jira.cornell.edu/browse/KFSPTS-24999.
Since the routing of the different documents has in many cases been customized for Cornell I am not sure we can use the XML provided by KualiCo but as a reference. I expect we need to start from the Cornell Specific workflow xml and add the new ObjectCode routing. I thought that we would have to first export the Cornell doc type, add the object code routing and then re-ingest. I tried this for the REQS doc type. Please let me know if you have some suggestion on how to better handle this or if you see any issue with the current plan. Thank you!

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:34]: I added the boundary type by string-appending. I could not determine what HttpCore 5 or HttpClient 5 helper classes I could use. If you have a tip on this I am happy to update.

I added the logging.

File: src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java:37]: We may want to ask KualiCo to update the superclass's version of this variable from private to protected.

File: src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/fp/document/datadictionary/DisbursementVoucherDocument.xml:151]: Thanks for the explanation. The follow-up ticket and current plan sound good, and if we uncover issues we can address them in that future ticket.

**Comment by cah292** [comment]
I'm still reviewing a few more things; I accidentally submitted my follow-up comments prematurely but they should still be valid. Sorry for the confusion.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:36]: We could add this header for now, but we'll need to make sure the punchout process is tested appropriately, and then make follow-up fixes as needed.

Also, I see that `MediaType` is a Spring-specific class, but the HTTP Core 5 javadocs mention that this `addHeader()` method just uses the value's `toString()` content, so I think this setup is okay as long as the string value is correct.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:37]: After looking further at the Spring `MediaType` class, I noticed that it has a two-arg constructor that accepts a base `MediaType` and a parameter map. If you wanted to simplify this further, you could use that two-arg constructor and pass in the "boundary" setting using a singleton map.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:44]: The old customization treats the request content as UTF-8 if no attachments are present, though it might not have changed the request's content type settings. Should this line be updated to set that charset on the request, or should we leave it as ISO-8859-1 instead?

**Review by cah292 — COMMENTED** [review]
I had a few more notes and suggestions that didn't get included in the other recent review. Other than that, the follow-up changes look good.

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:998]: updating

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:388]: updated

File: src/main/resources/edu/cornell/kfs/vnd/cu-spring-vnd.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/vnd/cu-spring-vnd.xml:35]: added

File: src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java:37]: I made a note in the relevant sub-task to report to KualiCo

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:44]: yes, I think this would make sense. Adding.

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:37]: updated

File: src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java:36]: I removed the accept header. I we find we need it we can add it back.

**Review by cah292 — COMMENTED** [review]
The code changes look good. This can be merged after the issues from the nonprod-sql PR have been addressed, as well as after we get the go-ahead to move this forward to DEV (depending on whether the functionals want us to hold off until closer to the end of the sprint).

Regarding the punchout-related handling in CuB2BDaoImpl, I'm okay with leaving out the "accept" header for the moment based on Daniela's feedback, but if we discover that we do need it then we can use a follow-up user story to implement the needed changes.

Also, given our inconsistencies with our setup of license header comments in CU-specific files, we may want a new user story to perform more cleanup of our files' license header setup.

**Comment by db483** [comment]
Thank you @cah292. I created https://jira.cornell.edu/browse/KFSPTS-26587 for the license header cleanup.

**Comment by db483** [comment]
This branch has been rebased after removing the spring security changes from develop.

**Review by cah292 — APPROVED** [review]
Code changes look good, unit tests pass and the related PRs are also ready. Merging.

**Pull request merged**
